controllers/kustomization_decryptor.go

```diff
@@ -731,3 +734,21 @@ func securePathErr(root string, err error) error {
 	}
 	return err
 }
+
+func formatForPath(path string) formats.Format {
+	switch {
+	case strings.HasSuffix(path, corev1.DockerConfigJsonKey):
+		return formats.Json
```

Comment:
Perhaps we should only look for this key when the _Secret_'s "type" field is "kubernetes.io/dockerconfigjson."

Reply:
Think that with the current approach, it gives people some flexibility to e.g. have opaque Secrets with multiple `.dockerconfigjson` files, for whatever reason that may be required. In addition to simplifying the code :-).